core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -793,17 +799,33 @@ public boolean isStoreWidth() {
         return storeWidth;
     }
 
+
+    public DataFlagEncoder setSpatialRules(int rules) {
+        this.spatialRules = rules;
+        return this;
+    }
+
+    public DataFlagEncoder setSpatialRuleLookup(SpatialRuleLookup spatialRuleLookup) {
+        if (spatialRuleLookup.size() != spatialRules)
+            throw new IllegalArgumentException("You have to configure the encoder to accept the identical amount of rules " +
+                    "that the spatial rule lookup has (" + spatialRuleLookup.size() + ") but it was " + spatialRules);
+
+        this.spatialRuleLookup = spatialRuleLookup;
+        return this;
+    }
+
     @Override
     protected String getPropertiesString() {
         return super.getPropertiesString() +
                 "|store_height=" + storeHeight +
                 "|store_weight=" + storeWeight +
-                "|store_width=" + storeWidth;
+                "|store_width=" + storeWidth +
+                "|spatial_rules=" + spatialRules;
```

Comment:
Should we safe the order and name of the countries? Or would this be too much overhead?